File: dotnet/EmitLogDirect.cs
using System;
using System.Linq;
using RabbitMQ.Client;

class EmitLogDirect {
    public static void Main(string[] args) {
        ConnectionFactory factory = new ConnectionFactory();
        factory.HostName = "localhost";
        using (IConnection connection = factory.CreateConnection())
        using (IModel channel = connection.CreateModel()) {
            channel.ExchangeDeclare("direct_logs", "direct");

            string severity = (args.Length > 0) ? args[0] : "info";
            string message = (args.Length > 1) ? string.Join(" ", args.Skip(1)
                                                                  .ToArray())
                                               : "Hello World!";
            byte[] body = System.Text.Encoding.UTF8.GetBytes(message);
            channel.BasicPublish("direct_logs", severity, null, body);
            Console.WriteLine(" [x] Sent '{0}':'{1}'", severity, message);
        }
    }
}

File: julia/emit_log_direct.jl
using AMQPClient
const VIRTUALHOST = "/"
const HOST = "127.0.0.1"


function send()
    # 1. Create a connection to the localhost or 127.0.0.1 of virtualhost '/'
    connection(; virtualhost=VIRTUALHOST, host=HOST) do conn
        # 2. Create a channel to send messages
        channel(conn, AMQPClient.UNUSED_CHANNEL, true) do chan
            # 3. Declare exchange
            exchange = "direct_logs"
            exchange_declare(chan, exchange, EXCHANGE_TYPE_DIRECT)
            # 4. Get severity and message
            if length(Base.ARGS) >= 3
                severity = Base.ARGS[1]
                received = join(Base.ARGS[2:end], ' ')
            else
                severity = "info"
                received = "Hello World"
            end
            data = convert(Vector{UInt8}, codeunits(received))
            msg = Message(data, content_type="text/plain", delivery_mode=PERSISTENT)
            # 5. Publish the message
            basic_publish(chan, msg; exchange=exchange, routing_key=severity)
            println("Message sent: $received, Severity: $severity")
        end
    end
end

send()
